File: src/main/java/de/tectoast/emolga/utils/ReplayAnalyser.java
package de.tectoast.emolga.utils;

import de.tectoast.emolga.utils.showdown.Player;

import java.util.List;
import java.util.Map;

@FunctionalInterface
public interface ReplayAnalyser {
    void analyse(Player[] game, String uid1, String uid2, List<Map<String, String>> kills, List<Map<String, String>> deaths, Object... optionalArgs);
}

File: src/main/kotlin/de/tectoast/emolga/utils/ReplayAnalyser.kt
package de.tectoast.emolga.utils

import de.tectoast.emolga.utils.showdown.Player

fun interface ReplayAnalyser {
    fun analyse(
        game: Array<Player>,
        uid1: String,
        uid2: String,
        kills: List<Map<String, String>>,
        deaths: List<Map<String, String>>,
        vararg optionalArgs: Any?
    )
}